cleverhans/loss.py

```diff
@@ -66,7 +66,8 @@ def fprop(self, x, y, **kwargs):
 
         logits = [self.model.get_logits(x, **kwargs) for x in x]
         loss = sum(
-            softmax_cross_entropy_with_logits(labels=y, logits=logit)
+            tf.reduce_mean(softmax_cross_entropy_with_logits(labels=y,
```

Comment:
Should we add a warning here alerting users of the change in behavior?

Reply:
I think the warning would be annoying, because even people who know about and want the new behavior would always see the warning.

Maybe one option would be to leave the existing class untouched, make a new class with the new behavior, and put a deprecation warning in the old class. e.g., put a deprecation warning in LossCrossEntropy, and make a new class called CrossEntropy with the new behavior. Are you OK with that?

Reply:
Agreed on the warning being potentially annoying. Your proposal sounds great, thanks!